lib/nodeJavaBridge.js

```diff
@@ -189,9 +194,21 @@ function isWritable(obj, prop) {
   return (Object.getOwnPropertyDescriptor(obj, prop) || {}).writable !== false;
 }
 
+var reservedPropertyNames = [ 'name', 'attributes', 'caller' ];
+
+function usableName(obj, name) {
+  // Node 0.10 and 0.12 have different behavior for whether function.name is declared writable or not.
+  // This test is constructed so that any unwritable property name OR the name 'name' will be altered,
+  // to give us consistent behavior for the one known problem case of properties named 'name'.
+  if (!isWritable(obj, name) || _.includes(reservedPropertyNames, name)) {
```

Comment:
@jsdevel I tried to explain it with the comment. Node 0.10 and 0.12 give different answers for which properties are writable. This means unit tests and our documentation will either have to take into account these differences, or we should use an implementation that gives consistent results for the known reserved words. I decided in favor of the latter.

One could argue that we should eliminate the redundancy by just doing the `reservedPropertyNames` check, as you did in your initial fix. But `isWritable()` gives us a solution that will continue to avoid an exception being thrown from java.import() even if in ES7+ there are new reserved words, and none of us proactively updates `reservedPropertyNames` for the new reserved words. 

I will squash before committing, and I'll defer that until tomorrow to give you a chance to respond, just in case you still think we should eliminate the redundancy.